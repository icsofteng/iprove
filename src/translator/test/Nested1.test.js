const {translate} = require('../../translator/z3')
const {translate: translate_latex} = require('../../translator/latex')

const test_constants = [{ value: 'P', varType: 'Bool' }, { value: 'Q', varType: 'Bool' }]
const test_rules = [
  {
    type: 'binary',
    symbol: 'and',
    lhs: {
      type: 'binary',
      symbol: 'and', 
      lhs: {
        type:'literal',
        value: 'P'
      }, 
      rhs: {
        type:'literal',
        value: 'Q'
      }
    },
    rhs: {
      type: 'literal',
      value: 'P'
    },
  }
]

test('Nested test 1', () => {
  expect(translate(test_rules, test_constants, [], [], [])).toMatchSnapshot()
})


test('Nested test latex', () => {
  expect(translate_latex(test_rules, test_constants, [], [], [])).toMatchSnapshot()
})