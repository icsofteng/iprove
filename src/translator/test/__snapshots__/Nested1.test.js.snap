// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nested test 1 1`] = `
"(declare-const P Bool)
(declare-const Q Bool)
(assert (not (and (and P Q) P)))
(check-sat)"
`;

exports[`Nested test latex 1`] = `
Array [
  "P \\\\land Q \\\\land P",
]
`;
